### PR TITLE
fix: Fixing 400 error in POST /platform/v1/spender/expenses/stats

### PR DESCRIPTION
**Pull request by neohacker18**
### Description
copilot:summary

copilot:poem

### Walkthrough
copilot:walkthrough

## Clickup
https://app.clickup.com/t/86cv9nqh2

### CODE REVIEW

**Comment by suyashpatil78** [comment]
@neohacker18 are we writing tests for this new changes; test workflow is failing because the branch is not covered in the tests.

File: src/app/fyle/my-expenses/my-expenses.page.ts [hunk 1]
```diff
@@ -335,12 +335,11 @@ export class MyExpensesPage implements OnInit {
...
         queryParams.state = 'in.(COMPLETE,DRAFT)';
 
-        if (queryParams['matched_corporate_card_transactions->0->corporate_card_number']) {
-          const cardParamsCopy = queryParams['matched_corporate_card_transactions->0->corporate_card_number'] as string;
-
-          queryParams.or = (queryParams.or || []) as string[];
-          queryParams.or.push('(matched_corporate_card_transactions->0->corporate_card_number.' + cardParamsCopy + ')');
-          delete queryParams['matched_corporate_card_transactions->0->corporate_card_number'];
+        if (queryParams.or) {
+          const hasState = Array.isArray(queryParams.or) && queryParams.or.some((element) => element.includes('state'));
```
Comment (suyashpatil78) [inline src/app/fyle/my-expenses/my-expenses.page.ts:339]: hasState -> `hasExpenseState`

**Comment by neohacker18** [comment]
> @neohacker18 are we writing tests for this new changes; test workflow is failing because the branch is not covered in the tests.

Have added a test case for this. The test workflow is green now. @suyashpatil78

**Review by suyashpatil78 — APPROVED** [review]
@neohacker18 please take a green from @Chethan-Fyle sir before merging

**Pull request merged**